src/ResourceManager/Kubernetes/Commands.Kubernetes/Commands/StartAzureRmKubernetesDashboard.cs

```diff
@@ -0,0 +1,330 @@
+            ValueFromPipeline =true,
+            HelpMessage ="A PSKubernetesCluster object, normally passed through the pipeline.")]
+        [ValidateNotNullOrEmpty]
+        public PSKubernetesCluster InputObject { get; set; }
+
+        /// <summary>
+        /// Cluster name
+        /// </summary>
+        [Parameter(Mandatory = true,
+            ParameterSetName = IdParameterSet,
+            Position = 0,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Id of a managed Kubernetes cluster")]
+        [ValidateNotNullOrEmpty]
+        public string Id { get; set; }
+
+        /// <summary>
+        /// Cluster name
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            Position = 0,
+            ParameterSetName = GroupNameParameterSet,
+            HelpMessage = "Name of your managed Kubernetes cluster")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Resource group name
+        /// </summary>
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = GroupNameParameterSet,
+            HelpMessage = "Resource group name")]
+        [ResourceGroupCompleter()]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = false,
+            HelpMessage = "Do not pop open a browser after establising the kubectl port-forward.")]
+        public SwitchParameter DisableBrowser { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public SwitchParameter PassThru { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            base.ExecuteCmdlet();
+
+            RunCmdLet(() =>
+            {
+                switch (ParameterSetName)
+                {
+                    case IdParameterSet:
+                    {
+                        var resource = new ResourceIdentifier(Id);
+                        ResourceGroupName = resource.ResourceGroupName;
+                        Name = resource.ResourceName;
+                        break;
+                    }
+                    case InputObjectParameterSet:
+                    {
+                        var resource = new ResourceIdentifier(InputObject.Id);
+                        ResourceGroupName = resource.ResourceGroupName;
+                        Name = resource.ResourceName;
+                        break;
+                    }
+                }
+
+                var tmpFileName = Path.GetTempFileName();
+                var encoded = Client.ManagedClusters.GetAccessProfiles(ResourceGroupName, Name, "clusterUser")
+                    .KubeConfig;
+                File.WriteAllText(
+                    tmpFileName,
+                    Encoding.UTF8.GetString(Convert.FromBase64String(encoded)));
+
+                var proxyUrl = "http://127.0.0.1:8001";
+
+                WriteVerbose(string.Format(
```

Comment:
We should likely look for kubectl on the path and write a nice error if it does not exist before oing further.

Reply:
Done. See GeneralUtilities.Probe for implementation.